### PR TITLE
Add docs

### LINKED ISSUE #22: Add documentation
We should really add docs, including design decisions, and intended use for different scenarios. Using ideas in #21 would be useful

### PR DESCRIPTION
**Pull request by charleskawczynski**
- Adds documentation.

Closes #22

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    AbstractEarthParameterSet <: AbstractParameterSet
+
+An earth parameter set, specific to planet Earth.
+"""
+abstract type AbstractEarthParameterSet <: AbstractParameterSet end
+
+"""
+    AbstractMicrophysicsParameterSet <: AbstractParameterSet
+
+An earth parameter set, specific to planet Earth.
```
Comment (trontrytel) [inline src/types.jl:20]:
```suggestion
A set of parameters for cloud and precipitation microphysics parameterisation.
```

File: src/types.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractMicrophysicsParameterSet <: AbstractParameterSet end
+
+"""
+    AbstractCloudParameterSet  <: AbstractMicrophysicsParameterSet
+
+A cloud parameter set, for atmospheric clouds
```
Comment (trontrytel) [inline src/types.jl:27]:
```suggestion
A set of parameters for cloud microphysics parameterisation.
```

File: src/types.jl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractCloudParameterSet  <: AbstractMicrophysicsParameterSet end
+
+"""
+    AbstractPrecipParameterSet <: AbstractMicrophysicsParameterSet
+
+A precipitation parameter set, for atmospheric precipitation
```
Comment (trontrytel) [inline src/types.jl:34]:
```suggestion
A set of parameters for precipitation microphysics parameterisation.
```

File: src/types.jl [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractPrecipParameterSet <: AbstractMicrophysicsParameterSet end
+
+"""
+    AbstractLiquidParameterSet <: AbstractCloudParameterSet
+
+A liquid parameter set, for liquid in the atmospheric
```
Comment (trontrytel) [inline src/types.jl:41]:
```suggestion
A set of parameters for cloud liquid water microphysics parameterisation. 
```

File: src/types.jl [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractLiquidParameterSet <: AbstractCloudParameterSet end
+
+"""
+    AbstractIceParameterSet    <: AbstractCloudParameterSet
+
+A ice parameter set, for ice in the atmosphere
```
Comment (trontrytel) [inline src/types.jl:48]:
```suggestion
A set of parameters for cloud ice microphysics parameterisation. 
```

File: src/types.jl [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractIceParameterSet    <: AbstractCloudParameterSet end
+
+"""
+    AbstractRainParameterSet   <: AbstractPrecipParameterSet
+
+A rain parameter set, for atmospheric rain
```
Comment (trontrytel) [inline src/types.jl:55]:
```suggestion
A set of parameters for rain microphysics parameterisation.
```

File: src/types.jl [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+"""
+abstract type AbstractRainParameterSet   <: AbstractPrecipParameterSet end
+
+"""
+    AbstractSnowParameterSet   <: AbstractPrecipParameterSet
+
+A snow parameter set, for snow in the atmosphere
```
Comment (trontrytel) [inline src/types.jl:62]:
```suggestion
A set of parameters for snow microphysics parameterisation. 
```

File: src/types.jl [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+An earth parameter set, specific to planet Earth.
+"""
+abstract type AbstractMicrophysicsParameterSet <: AbstractParameterSet end
+
+"""
+    AbstractCloudParameterSet  <: AbstractMicrophysicsParameterSet
+
+A cloud parameter set, for atmospheric clouds
+"""
```
Comment (trontrytel) [inline src/types.jl:28]: Should we also say in the description that they consist of a subset of microphysics parameters, etc?

File: docs/src/API.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+Atmos.SubgridScale.c_e2_KASM
+Atmos.SubgridScale.c_1_KASM
+Atmos.SubgridScale.c_2_KASM
+Atmos.SubgridScale.c_3_KASM
+```
+
+### Microphysics
+
+```@docs
+Atmos.Microphysics.n0
```
Comment (trontrytel) [inline docs/src/API.md:111]: We could also link the relevant ClimateMachine docs somewhere? A huge chunk of the Microphysics docs in the Machine give references to why certain parameters have certain values

File: src/types.jl (hunk 8 above)
Comment (charleskawczynski) [reply src/types.jl:28]: The structure of the concrete type is flexible from the perspective of CLIMAParameters, so I don't think we need to include this information.

File: docs/src/API.md (hunk 9 above)
Comment (charleskawczynski) [reply docs/src/API.md:111]: Hmmm, good question. I think adding a link would be a good idea, yes. I'll add one to the microphysics theory docs.

Comment (trontrytel) [reply docs/src/API.md:111]: Or not. We could also document here the way the Parameters are meant to be used and keep the theory to the Machine docs. I don't know which is better. 

But maybe a link somewhere up top to the Machine docs would be useful

Comment (charleskawczynski) [reply docs/src/API.md:113]: I added a small blurb to the microphysics header. Let me know what you think.

**Comment by charleskawczynski** [comment]
> Looks fine. We don't really have a writeup anywhere, describing the _why_ of this module. Maybe something for the landing page, at some point?

I think Simon's issue has some useful stuff, that I thought about including in the docs. But I agree that having a more comprehensive design philosophy is a good idea to add eventually.

**Pull request merged**